LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/ClientCertificateValidatorService.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+namespace LoRaWan.NetworkServer.BasicsStation
+{
+    using System;
+    using System.Linq;
+    using System.Net.Security;
+    using System.Security.Cryptography.X509Certificates;
+    using System.Text.RegularExpressions;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.Logging;
+
+    internal class ClientCertificateValidatorService
+    {
+        private readonly IBasicsStationConfigurationService stationConfigurationService;
+        private readonly ILogger<ClientCertificateValidatorService> logger;
+
+        public ClientCertificateValidatorService(IBasicsStationConfigurationService stationConfigurationService,
+                                                 ILogger<ClientCertificateValidatorService> logger)
+        {
+            this.stationConfigurationService = stationConfigurationService;
+            this.logger = logger;
+        }
+
+        // Following synchronous implementation is needed for ASP.NET Core ClientCertificateValidation
+        public virtual bool Validate(X509Certificate2 certificate, X509Chain? chain, SslPolicyErrors sslPolicyErrors) =>
+            ValidateAsync(certificate, chain, sslPolicyErrors, default).GetAwaiter().GetResult();
+
+        public virtual async Task<bool> ValidateAsync(X509Certificate2 certificate, X509Chain? chain, SslPolicyErrors sslPolicyErrors, CancellationToken token)
```

Comment:
A very open-ended question that I think will be resolved at the end: we don't expect this to happen often, right? So, async validation that we will block on the call-site is not that big of an issue, right? Do we have any alternatives that we considered, though?

Reply:
Unfortunately there are no other alternatives, since the Client Certificate validation is happening synchronously on ASP.NET side. We tried to find alternatives with Atif, but we concluded that it's just fine to do this check synchronously for now